src/reflect/value.go

```diff
@@ -948,7 +948,15 @@ func (v Value) SetCap(n int) {
 }
 
 func (v Value) SetLen(n int) {
-	panic("unimplemented: (reflect.Value).SetLen()")
+	if v.typecode.Kind() != Slice {
+		panic(&ValueError{"reflect.Value.SetLen", v.Kind()})
+	}
+
+	hdr := (*sliceHeader)(v.value)
+	if uintptr(n) > hdr.cap {
```

Comment:
There is another small issue here on AVR (with `SetLen(0x10005)`), would be nice if it works correctly on those chips too.

```suggestion
	if int(uintptr(n)) != n || uintptr(n) > hdr.cap {
```